src/script_opt/CPP/InitsInfo.h

```diff
@@ -258,9 +258,8 @@ class CPP_CompoundInitsInfo : public CPP_InitsInfo
 class CPP_InitInfo
 	{
 public:
-	// No constructor - basic initialization happens when the object is
-	// added via AddInstance() to a CPP_InitsInfo object, which in turn
-	// will lead to invocation of this object's SetOffset() method.
+	CPP_InitInfo(const IntrusivePtr<Obj>& _o) : o(_o.get()) { }
```

Comment:
This is without context at all, but mostly wondering if the `o` member could/should be a `zeek::IntrusivePtr<Obj>` rather than a raw pointer. Otherwise the pointer may be invalidated. It is possible to null an intrusive ptr, but maybe I'm missing something else.

Reply:
Hmm, I may have understood. This may also receive `const ID*` rather than a `IDPtr` and that's difficult to convert to an IntrusivePtr.